Tests/UnitTests/PaymentsCoreTests/NetworkingTests/APIClient_Tests.swift

```diff
@@ -0,0 +1,189 @@
+import XCTest
+@testable import PaymentsCore
+
+class APIClient_Tests: XCTestCase {
+
+    lazy var apiClient: APIClient = {
+        APIClient(
+            urlSession: URLSession(urlProtocol: URLProtocolMock.self),
+            environment: .sandbox
+        )
+    }()
+
+    func testFetch_withAccessTokenSuccessMockResponse_returnsValidAccessToken() {
```

Comment:
Another nitpicky naming comment. We can make the naming format the same for all tests. Currently some say "with" some don't. Some say "returns" some say "vends". Long test names is OK and preferred.